src/Chart.php

```diff
@@ -112,68 +156,80 @@ public function setOptions(array $options)
         return $this;
     }
 
+    /**
+     * @param array<string, array<mixed, mixed>> $options column_name => array of options
+     *
+     * @return $this
+     */
+    public function setColumnOptions(array $options)
+    {
+        // IMPORTANT: use replace not merge here to preserve numeric keys !!!
+        $this->columnOptions = array_replace_recursive($this->columnOptions, $options);
+
+        return $this;
+    }
+
     /**
      * Specify data source for this chart. The column must contain
-     * the textual column first followed by sumber of data columns:
+     * the textual column first followed by number of data columns:
      * setModel($month_report, ['month', 'total_sales', 'total_purchases']);.
      *
      * This component will automatically figure out name of the chart,
      * series titles based on column captions etc.
      *
-     * Example for bar chart with two side-by side bars per category, and one of them stacked:
-     *
-     * $chart->setModel(
-     *      $model,
-     *      ['month', 'turnover_month_shoes', 'turnover_month_shirts', 'turnover_month_trousers', 'turnover_month_total_last_year'],
-     *      [1, 1, 1, 2] // 1 => shoes+shirts+trousers, 2 => total last year
-     *  );
-     *
      * @param array<int, string> $columns
-     * @param array<int, mixed>  $stacks
      */
-    public function setModel(Model $model, array $columns = [], array $stacks = []): void
+    public function setModel(Model $model, array $columns = []): void
     {
         if ($columns === []) {
             throw new Exception('Second argument must be specified to Chart::setModel()');
         }
+        $this->columns = $columns;
+
+        parent::setModel($model);
+
+        $this->prepareDatasets();
+    }
+
+    /**
+     * Fills dataset with data from data model.
+     */
+    protected function prepareDatasets(): void
+    {
+        if ($this->model === null || $this->columns === null) {
```

Comment:
Is model required to be set or not?

Reply:
Yes model and columns are required. Do you think exception here would be better?